subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/TaskCacheabilityReasonIntegrationTest.groovy

```diff
@@ -342,6 +374,39 @@ class TaskCacheabilityReasonIntegrationTest extends AbstractIntegrationSpec impl
         assertCachingDisabledFor VALIDATION_FAILURE, "Caching has been disabled to ensure correctness. Please consult deprecation warnings for more details.", ":consumer"
     }
 
+    def "cacheability for a cacheable task can be disabled via #condition"() {
```

Comment:
Didn't we have tests for those conditions?

Reply:
FTR, we discussed that we need integration tests for these cases, so we'll keep these tests here.